Fix roast init to create workflows in roast/ directory

ersync opened the pull request:
- roast init now creates roast/ directory if it doesn't exist
- Copies examples into roast/ subdirectory instead of current directory
- Makes roast init consistent with roast list command

Fixes #396

ersync commented:
I have signed the CLA!

dersam reviewed (changes requested):
Looks good, thanks! CI failures are style linting related, you can autocorrect them with `rake rubocop`.

ersync commented:
> Looks good, thanks! CI failures are style linting related, you can autocorrect them with `rake rubocop`.

Hi @dersam, thanks for the review! I've addressed the linting issues with `rake rubocop` and updated the branch. Please let me know if anything else needs attention.
